prusti-encoder/src/encoders/type/domain.rs

```diff
@@ -136,7 +136,7 @@ impl TaskEncoder for DomainEnc {
                 Ok((enc.finalize(), DomainEncSpecifics::Param))
             }
             TyKind::Adt(adt, params) => {
-                let base_name = vcx.tcx.item_name(adt.did()).to_ident_string();
+                let base_name = vcx.tcx().item_name(adt.did()).to_ident_string();
```

Comment:
Why is this a method now?

Reply:
The `VirCtxt` may not contain a `TyCtxt`, when running in the server (that's a standalone program, separate from `rustc`), or when running tests. So `tcx`, the field, is now an `Option`, and this method does this to avoid `unwrap`s or similar everywhere. However, as you pointed out, `VirCtxt` can also have a `Deref` implementation to avoid this completely at callsites.

Reply:
> The VirCtxt may not contain a TyCtxt, when running in the server (that's a standalone program, separate from rustc), or when running tests

An alternative reading of this is that the `VirCtxt` is lumping two separate things together: I would assume that the server / test runner is doing different things (ie: lowering VIR to JNI) compared to the frontend (lowering MIR to VIR) and thus perhaps they should in actuality be two separate contexts.